src/v/kafka/server/quota_manager.cc

```diff
@@ -285,15 +351,35 @@ ss::future<throttle_delay> quota_manager::throttle_fetch_tp(
 
 // erase inactive tracked quotas. windows are considered inactive if
 // they have not received any updates in ten window's worth of time.
-void quota_manager::gc(clock::duration full_window) {
-    auto now = clock::now();
-    auto expire_age = full_window * 10;
-    // c++20: replace with std::erase_if
-    absl::erase_if(
-      _client_quotas,
-      [now, expire_age](const std::pair<ss::sstring, client_quota>& q) {
-          return (now - q.second.last_seen) > expire_age;
-      });
+void quota_manager::gc() {
+    auto full_window = _default_num_windows() * _default_window_width();
+    ssx::background
+      = ssx::spawn_with_gate_then(
+          _gate,
+          [full_window,
+           _client_quotas = std::ref(_client_quotas)]() -> ss::future<void> {
+              return ss::smp::submit_to(
+                _client_quotas.get().shard_id(),
+                [full_window, _client_quotas = std::ref(_client_quotas)]()
+                  -> ss::future<void> {
```

Comment:
I don't think it makes sense for every shard to dispatch a gc to the single shard. The tiimer can be set just on the worker shard,

Reply:
Changing this to only fire from the owner shard